Add support for LIKE and STARTS_WITH operators

edhollinghurst opened the pull request:
This PR adds support for use of the LIKE and STARTS_WITH operators when adding a field condition to a query.

edhollinghurst committed:
Update RestfulBase.php

Allow LIKE and STARTS_WITH operators to be used.

edhollinghurst committed:
Update RestfulDataProviderEFQ.php

Add handling for LIKE and STARTS_WITH operators.

amitaibu commented:
Thanks. This will also require a (simple) simpleTest :)

e0ipso commented:
We also need support for those operators on entity properties, not just fields.

edhollinghurst committed:
Update RestfulDataProviderEFQ.php

The extra handling for LIKE and STARTS_WITH actually isn't needed as this is already handled correctly for both fields and properties on lines 179 and 185 respectively.

edhollinghurst committed:
Update RestfulListTestCase.test

Added tests for use of LIKE and STARTS_WITH operators.

edhollinghurst commented:
When I looked at this again I realised that the LIKE and STARTS_WITH operators are already handled correctly in RestfulDataProviderEFQ.php for both fields and properties. It's just a case of enabling these operators in RestfulBase.php.

I've added a couple of tests for these operators to RestfulListTestCase.test.

e0ipso commented:
@edhollinghurst actually by enabling it in `RestfulBase` you are enabling it as well for all of the data providers. Probably this will fail for other data providers, like for instance `RestfulDataProviderDbQuery`.

edhollinghurst committed:
Moving the addition of the 'LIKE' and 'STARTS_WITH' operators from the RestfulBase class to the RestfulDataProviderEFQ class.

edhollinghurst commented:
I've now taken out the change from `RestfulBase` and overridden the function in `RestfulDataProviderEFQ`.

I also made a small correction to the error message that's thrown if the operator is not allowed. `$operators` was being passed as the token when this should be `$operator`.

edhollinghurst commented:
@mateu-aguilo-bosch Is there anything more you need me to do in terms of this?

amitaibu commented on plugins/restful/RestfulDataProviderEFQ.php:
```diff
@@ -209,6 +209,34 @@ protected function getColumnFromProperty($property_name) {
+      '<=',
+      '<>',
+      '!=',
+      'IN',
```
There's also

```
+      'CONTAINS',
+      'IN',
+      'NOT IN',
+      'STARTS_WITH',
```

btw, the file chmod should be changed back to 644

amitaibu commented:
Should we also add this to `RestfulBase.php`?

This needs a re-roll.

e0ipso commented:
> Should we also add this to RestfulBase.php?

I don't think that we should. Those apply to all of the data providers, but only some of them support that list of operators.

e0ipso commented:
I'm going to close this issue to unclutter the PR view. Feel free to re-open if you pick up this work again.

Thanks for everything!
